File: src/main/java/santiagoAndFerdy/vgs/user/IUser.java
package santiagoAndFerdy.vgs.user;

import santiagoAndFerdy.vgs.discovery.IAddressable;
import santiagoAndFerdy.vgs.model.Job;

import java.net.MalformedURLException;
import java.rmi.NotBoundException;
import java.rmi.Remote;
import java.rmi.RemoteException;

/**
 * Created by Fydio on 4/3/16.
 */
public interface IUser extends Remote {
    void acceptResult(Job j) throws RemoteException;
    void createJobs(int rmId, int ms, int n) throws RemoteException;
}

File: src/main/scala/grid/user/IUser.scala
package grid.user

import java.rmi.RemoteException

import grid.cluster.Addressable
import grid.model.Job

import scala.concurrent.duration.Duration

trait IUser extends Addressable {

  @throws(classOf[RemoteException])
  def createJobs(rmId: Int, n: Int, ms: Int): Unit

  @throws(classOf[RemoteException])
  def acceptJob(job: Job): Unit

  @throws(classOf[RemoteException])
  def acceptResult(job: Job): Unit
}
